Corona-Warn-App/src/main/java/de/rki/coronawarnapp/storage/LocalData.kt

```diff
@@ -383,76 +382,29 @@ object LocalData {
      * SERVER FETCH DATA
      ****************************************************/
 
-    /**
-     * Gets the last time the server fetched the diagnosis keys from the server as Date object
-     * from the EncryptedSharedPrefs
-     *
-     * @return timestamp as Date
-     */
-    // TODO should be changed to Long as well to align with other timestamps
-    fun lastTimeDiagnosisKeysFromServerFetch(): Date? {
-        val time = getSharedPreferenceInstance().getLong(
-            CoronaWarnApplication.getAppContext()
-                .getString(R.string.preference_timestamp_diagnosis_keys_fetch),
-            0L
-        )
-        if (time == 0L) return null
-
-        return Date(time)
-    }
-
-    fun lastTimeDiagnosisKeysFromServerFetchFlow() =
+    val lastTimeDiagnosisKeysFetchedFlowPref by lazy {
         getSharedPreferenceInstance()
-            .createFlowPreference<Long?>(CoronaWarnApplication.getAppContext()
-                .getString(R.string.preference_timestamp_diagnosis_keys_fetch), 0L).flow
-            .map { if (it != null && it != 0L) Date(it) else null }
+            .createFlowPreference(key = "preference_timestamp_diagnosis_keys_fetch", 0L)
+    }
 
-    /**
-     * Sets the last time the server fetched the diagnosis keys from the server as Date object
-     * from the EncryptedSharedPrefs
-     *
-     * @param value timestamp as Date
-     */
-    fun lastTimeDiagnosisKeysFromServerFetch(value: Date?) {
-        getSharedPreferenceInstance().edit(true) {
-            putLong(
-                CoronaWarnApplication.getAppContext()
-                    .getString(R.string.preference_timestamp_diagnosis_keys_fetch),
-                value?.time ?: 0L
-            )
+    var lastTimeDiagnosisKeysFromServerFetch: Date?
+        get() = lastTimeDiagnosisKeysFetchedFlowPref.value.let {
+            if (it != 0L) Date(it) else null
         }
-    }
```

Comment:
unrelated to the bug; i don't think this the right time for refactoring (including regression bugs from it)

Reply:
This is not refactoring, but a bugfix for:

> We were using a mix of different preferences objects when updating the last download timestamp which could cause them to go out of sync.

Reply:
See new changes. Hope it's okay now.

Reply:
but here one preference only was used: key = getString(R.string.preference_timestamp_diagnosis_keys_fetch), you just made it a flow, or am i missing something?